### PR TITLE
Fixed shiva taking damage when in stone

### PR DESCRIPTION
**Pull request by TokyoSU**
- GetTargetOnLOS() was not using obj->HitRoutine when shooting enemies which was not triggered but drawed.

### CODE REVIEW

File: TombEngine/Game/control/los.cpp [hunk 1]
```diff
@@ -183,32 +183,16 @@ bool GetTargetOnLOS(GameVector* origin, GameVector* target, bool drawTarget, boo
...
 							{
 								if (item->ObjectNumber >= ID_SMASH_OBJECT1 && item->ObjectNumber <= ID_SMASH_OBJECT8)
+								{
 									SmashObject(itemNumber);
+								}
 								else
 								{
-									switch (object->hitEffect)
-									{
-									case HitEffect::Blood:
```
Comment (Kubsy) [inline TombEngine/Game/control/los.cpp:191]: Why was the case switch for effects removed? Is it not needed?

File: TombEngine/Game/control/los.cpp [hunk 2]
```diff
@@ -183,32 +183,16 @@ bool GetTargetOnLOS(GameVector* origin, GameVector* target, bool drawTarget, boo
 							else
 							{
 								if (item->ObjectNumber >= ID_SMASH_OBJECT1 && item->ObjectNumber <= ID_SMASH_OBJECT8)
+								{
 									SmashObject(itemNumber);
+								}
 								else
 								{
-									switch (object->hitEffect)
```
Comment (Kubsy) [inline TombEngine/Game/control/los.cpp:189]: Why was the case switch for effects removed? Is it not needed?

**Review by Sezzary — APPROVED** [review]

File: TombEngine/Game/control/los.cpp (hunk 2 above)
Comment (TokyoSU) [reply TombEngine/Game/control/los.cpp:189]: DefaultItemHit do that itself :)

**Comment by Adngel** [comment]
This is not an issue of this branch, but I wonder, is possible use it to fix two things?

- The shiva statue is not moving down 1 click steps. (So I trigger it on a 1 click "pedestal" but it gets like locked in that sector).
- The shiva statue is not getting any damage from explossions.

These issues also happened in develop branch.

**Comment by TokyoSU** [comment]
> This is not an issue of this branch, but I wonder, is possible use it to fix two things?
> 
> * The shiva statue is not moving down 1 click steps. (So I trigger it on a 1 click "pedestal" but it gets like locked, walking in rounds in that sector).
> * The shiva statue is not getting any damage from explossions.
> 
> These issues also happened in develop branch.

Fixed :)

**Review by Adngel — APPROVED** [review]

**Review by Kubsy — APPROVED** [review]

**Pull request merged**